### ISSUE TITLE
TRoW S15 A New Land: Allow Lord Typhon to transfer or lend his storm trident to another merman when he leaves

### ISSUE THREAD

**Issue by max-torch** [wesnoth/wesnoth#6103]
### Game and System Information
 - 1.15.17
 - Steam
 - Windows 10
 - In The Rise of Wesnoth Campaign

### The problem
In S13 Peoples in Decline, there is a storm trident that can be picked up by any merman. If Lord Typhon picks it up its potential is wasted because at the end of S15 he just leaves along with the storm trident. This trident is an item that could be useful in the next several scenarios and reaches a peak of importance in S20 The Return of the Fleet because there is an enemy leader on an island easily reachable by merfolk. It is unfair to the player to suddenly lose the trident through no fault of their own because they could not anticipate that it would be wasted if given to Lord Typhon.

### The Desired Feature
Either don't allow Lord Typhon to be the one to pick up the trident in S13, or allow Lord Typhon to choose someone else to give it to before he leaves.

**Comment by Wedge009** [comment]
I agree this situation (where we lose Lord Typhon possibly without expecting it) is annoying, but I observe this kind of thing even in commercial games sometimes. I'm not sure how it could be implemented, plus one would have to account for the (slim) possibility that there's no eligible unit (other mer?) to pass the trident to.

It's a nice-to-have, though.

**Comment by max-torch** [comment]
Some Pseudocode
```python
Trident_check_func():
    if (Lord Typhon has Trident) AND (Recall List contains Merfolk):
        Lord Typhon says "someone else can have the trident"
        Show_Dialog_menu_for_selecting_merfolk_from_recall_list()
    else:
        return
```

**Comment by Wedge009** [comment]
I meant in the context of WML, the logic isn't difficult. I think any modifications applied to Typhon would need to be transferred when he departs, while also writing the dialogue in a way that didn't feel jarring and awkward.
https://github.com/wesnoth/wesnoth/blob/5f3382e8d8928f14aec0de835bf7d9fc7d5bedb2/data/campaigns/The_Rise_Of_Wesnoth/scenarios/15_A_New_Land.cfg#L479-L498

This kind of thing might be suitable for an easier campaign like Heir to the Throne where references to game-play elements are plentiful - especially in the earlier scenarios as different unit types are introduced to Konrad's recruit roster - and act as a gentle transition from explicit tutorial to in-universe story exposition. But I don't think any such thing exists in The Rise of Wesnoth. Perhaps if any transition of the trident modification was to occur it should only apply in easy difficulty as a concession.

**Comment by max-torch** [comment]
Interesting to see that further merman recruits was allowed if playing on easy. Anyhow, regardless of difficulty, my argument is about how this issue affects the fairness of the campaign, which could also could be equal in importance to the challenge of the campaign.

**Comment by stevecotton** [comment]
The code for picking up the trident (in `data/core/macros/items.cfg`) already checks that the unit doesn't already have a storm trident. On easy, maybe Lord Typhon should already have one when he joins in S12.

**Comment by Wedge009** [comment]
I suppose that's a possible work-around. Maintainer @ln-zookeeper doesn't appear to be around any more, I wonder if @nemaara has any thoughts on this possibly making things too easy for a campaign classified as 'Hard'.

From a wider perspective, surprises like this often happen in campaigns. So I'm not sure how much effort we should go to coddle the player into avoiding situations like these - although I'll grant it makes sense to give leaders or hero units bonuses to better defend themselves.

**Comment by nemaara** [comment]
Maybe the way to go would be to give Lord Triton one to begin with and then disallow him from taking it. Any of the above sound fine to me, the storm trident isn't going to affect balancing a crazy amount.

**Comment by Wedge009** [comment]
Lord Typhon is introduced with his escorts in S12:
https://github.com/wesnoth/wesnoth/blob/e398846ccbf21375c95c76fd73b5f57445f26341/data/campaigns/The_Rise_Of_Wesnoth/scenarios/12_A_Final_Spring.cfg#L388-L390

The trident is a standard item:
https://github.com/wesnoth/wesnoth/blob/e398846ccbf21375c95c76fd73b5f57445f26341/data/core/macros/items.cfg#L514-L552

I'm not sure how to declare Typhon such that he already has one. Typhon seems to be declared as a standard Merman Warrior - how to say he `[has_attack]name="storm trident"[/has_attack]`?

**Comment by Wedge009** [comment]
The Rise of Wesnoth factors out hero unit creation here:
https://github.com/wesnoth/wesnoth/blob/e398846ccbf21375c95c76fd73b5f57445f26341/data/campaigns/The_Rise_Of_Wesnoth/utils/trow-macros.cfg#L12-L27

I thought it possible to have a similar entry specifically for Lord Typhon but with the storm trident already in his possession, but I couldn't get it to work as intended. Looking throughout the campaigns storm tridents are only ever declared as items to be obtained in a given scenario, I couldn't find any instance where someone already has a storm trident already equipped - so I don't have an example to work from.

Another possibility I can think of is to declare Typhon as a special case of Merman Warrior, with the storm trident already as one of his attacks - I assume that should disallow him from picking up any storm tridents that happen to be lying around. I think this might be a little more work, though - would have to make a scenario-specific unit declaration for Typhon,

Again, I haven't dealt with WML much so I'm not sure if there's a 'correct' way to achieve what we want.

**Comment by Wedge009** [comment]
Closed via #6113 (thanks!)
